inotify_test.go

```diff
@@ -453,3 +453,52 @@ func TestInotifyOverflow(t *testing.T) {
 			numDirs*numFiles, creates)
 	}
 }
+
+func TestInotifyDeleteOpenedFile(t *testing.T) {
+	testDir := tempMkdir(t)
+	defer os.RemoveAll(testDir)
+
+	testFile := filepath.Join(testDir, "testfile")
+
+	// create and open a file
+	fd, err := os.Create(testFile)
+	if err != nil {
+		t.Fatalf("Create failed: %v", err)
+	}
+	defer fd.Close()
+
+	w, err := NewWatcher()
+	if err != nil {
+		t.Fatalf("Failed to create watcher: %v", err)
+	}
+	defer w.Close()
```

Comment:
I would make sure to that the watcher has been closed, to be consistent with other tests:
```go
defer func() {
  w.Close()
  // Wait for the close to complete.
  time.Sleep(50 * time.Millisecond)
  isWatcherReallyClosed(t, w)
}()
```

Reply:
1. This check is present in test cases which test that the watcher is closed; here's the list of such tests:
* `TestInotifyCloseRightAway`
* `TestInotifyCloseSlightlyLater`
* `TestInotifyCloseSlightlyLaterWithWatch`
* `TestInotifyCloseAfterRead`

In other test cases, including this one, we're not testing this particular aspect, so we don't care if the watcher is closed.

2. Calling `t.Fatalf()` (which `isWatcherReallyClosed()` might do) from a goroutine won't have the desired effect (as `t.Fatalf()` should be called from the same goroutine as the test), so adding this would be useless.